Temperature: add triggers and conditions (Labs feature) (#45229)
home-assistant/home-assistant.github.io: source/_triggers/temperature.crossed_threshold.markdown

10. Select **Save**.

### Options in the UI

{% options_ui %}
Threshold type:
  description: |
    Controls the zone the reading must enter for the trigger to fire:

    - **Above** or **Below**: enter a value to fire when the reading crosses that level.
    - **In range**: enter a lower and upper bound to fire when the reading enters the range from outside.
    - **Outside range**: enter a lower and upper bound to fire when the reading leaves the range (crosses past either bound).

    For each mode you can enter a fixed temperature or reference a sensor entity or [number helper](/integrations/input_number/) entity.
  required: true
Unit:
  description: The temperature unit to use for threshold comparison. Accepts `°C` or `°F`. Required when using numerical thresholds (not required when using entity references). Default is `°C`.
  required: false
Trigger when:
  description: |
    When multiple entities are targeted, controls when the trigger fires:

    - **Each**: fire every time any targeted entity crosses the threshold.
    - **First**: fire only on the first crossing.

>     - **All**: fire only after every targeted entity crosses the threshold.

Reviewer: Same as above:
```suggestion
    - **Each**: fires every time any targeted entity crosses the threshold.
    - **First**: fires only on the first crossing.
    - **All**: fires only after every targeted entity crosses the threshold.
```

    - `type: above` or `type: below`: Provide `value` with a `number` key (for a literal number) or an `entity` key (for an `input_number`, `number`, or `sensor` entity)
    - `type: between` or `type: outside`: Provide `value_min` and `value_max`, each with a `number` key (for a literal number) or an `entity` key (for an `input_number`, `number`, or `sensor` entity)

    When using the `number` key, you must also include `unit_of_measurement` to specify the temperature unit (`°C` or `°F`). When using the `entity` key, the unit is taken from the entity itself.

    For example:

    ```yaml
    threshold:
      type: between
      value_min:
        number: 18
        unit_of_measurement: °C
      value_max:
        entity: input_number.max_comfort_temperature
    ```

    A `sensor` entity's current reading is used as the threshold, which lets you compare two temperature readings dynamically.
  required: true
  type: map
behavior:
  description: |
    When multiple entities are targeted, controls when the trigger fires. Accepts:

    - `any`: fire every time any targeted entity crosses the threshold.
    - `first`: fire only on the first crossing.

>     - `last`: fire only after every targeted entity crosses the threshold.

Reviewer: Same as above:
```suggestion
    - `any`: fires every time any targeted entity crosses the threshold.
    - `first`: fires only on the first crossing.
    - `last`: fires only after every targeted entity crosses the threshold.
```

- The trigger works with [climate](/integrations/climate/) entities, [water heater](/integrations/water_heater/) entities, [weather](/integrations/weather/) entities, and sensors with the temperature device class.
- Climate, water heater, and weather entities that don't report a current temperature attribute are automatically excluded from the trigger. Only entities with a valid temperature value can fire the trigger.
- All temperature values are automatically converted to the unit you specify. For example, if your sensor reports in Fahrenheit but you configure the trigger in Celsius, the conversion happens automatically.

{% include triggers/try_it.md %}

{% include triggers/more_examples.md %}

### Automation: turn off climate when temperature enters comfort range

This automation turns off the living room climate system the moment the temperature crosses into the comfort range (20 to 22°C), saving energy once comfortable conditions are achieved.

- **Trigger**: Temperature crossed threshold
- **Target**: Living room temperature sensor
- **Threshold type**: In range (20-22°C)

> - **Action**: Climate: Set HVAC mode

Reviewer: ```suggestion
- **Action**: Set thermostat HVAC mode (state: off)
```

{% details "YAML example for turning off climate when comfortable" %}

{% example %}
automation: |
  alias: "Turn off climate when living room is comfortable"
  triggers:
    - trigger: temperature.crossed_threshold
      target:
        entity_id: sensor.living_room_temperature
      options:
        threshold:
          type: between
          value_min:
            number: 20
            unit_of_measurement: "°C"
          value_max:
            number: 22
            unit_of_measurement: "°C"
  actions:
    - action: climate.set_hvac_mode
      target:
        entity_id: climate.living_room
      data:
        hvac_mode: "off"
{% endexample %}

{% enddetails %}

### Automation: alert when bedroom temperature enters comfort range

After opening windows to cool down a stuffy bedroom, this automation alerts you the moment the temperature enters your preferred comfort range so you can close the windows.

- **Trigger**: Temperature crossed threshold
- **Target**: Bedroom temperature sensor
- **Threshold type**: In range (20-22°C)

> - **Action**: Notify: Send notification

Reviewer: ```suggestion
- **Action**: Send a notification
```

{% details "YAML example for comfort range entry alert" %}

{% example %}
automation: |
  alias: "Alert when bedroom temperature is comfortable"
  triggers:
    - trigger: temperature.crossed_threshold
      target:
        entity_id: sensor.bedroom_temperature
      options:
        threshold:
          type: between
          value_min:
            number: 20
            unit_of_measurement: "°C"
          value_max:
            number: 22
            unit_of_measurement: "°C"
  actions:
    - action: notify.mobile_app
      data:
        message: >-
          Bedroom temperature is now comfortable at
          {{ trigger.to_state.state }}°C
{% endexample %}

{% enddetails %}

### Automation: prevent false triggers with a delay

To avoid false triggers from brief temperature fluctuations when opening a door or window, add a **For at least** delay. This automation only fires after the temperature has been below 18°C for 5 minutes.

- **Trigger**: Temperature crossed threshold
- **Target**: Living room temperature sensor

> - **Threshold type**: Below 18°C

Reviewer: ```suggestion
- **Threshold type**: Below (18°C)
```

- **For at least**: 5 minutes

> - **Action**: Climate: Set HVAC mode

Reviewer: ```suggestion
- **Action**: Set thermostat HVAC mode (state: heat)
```

{% details "YAML example with delay to prevent false triggers" %}

{% example %}
automation: |
  alias: "Turn on heating when consistently cold"
  triggers:
    - trigger: temperature.crossed_threshold
      target:
        entity_id: sensor.living_room_temperature
      options:
        threshold:
          type: below
          value:
            number: 18
            unit_of_measurement: "°C"
        for: "00:05:00"
  actions:
    - action: climate.set_hvac_mode
      target:
        entity_id: climate.living_room
      data:
        hvac_mode: heat
{% endexample %}

{% enddetails %}

### Automation: trigger heating based on adjustable comfort temperature

Trigger the heating when temperature crosses below your personal comfort threshold. Use a number helper as the threshold so you can easily adjust it through the UI without editing the automation.

- **Trigger**: Temperature crossed threshold
- **Target**: Living room temperature sensor
- **Threshold type**: Below (entity: comfort temperature threshold)

> - **Action**: Climate: Set HVAC mode

Reviewer: ```suggestion
- **Action**: Set thermostat HVAC mode (state: heat)
```

      value_max:
        number: 22
        unit_of_measurement: "°C"
{% endexample %}

### Options in YAML

YAML sometimes provides additional options for more complex use cases that are not available through the UI.

{% options_yaml %}
threshold:
  description: |
    A mapping that defines the zone the reading must enter for the trigger to fire:

    - `type: above` or `type: below`: Provide `value` with a `number` key (for a literal number) or an `entity` key (for an `input_number`, `number`, or `sensor` entity)
    - `type: between` or `type: outside`: Provide `value_min` and `value_max`, each with a `number` key (for a literal number) or an `entity` key (for an `input_number`, `number`, or `sensor` entity)

    For example:

    ```yaml
    threshold:
      type: between
      value_min:
        number: 18
        unit_of_measurement: °C
      value_max:
        entity: input_number.max_comfort_temperature
    ```

    A `sensor` entity's current reading is used as the threshold, which lets you compare two temperature readings dynamically.
  required: true
  type: map
unit:
  description: >
    The temperature unit to use for threshold comparison. Accepts `°C` or `°F`.

>   required: true

Author: good catch! The unit doesn't actually even exist. there is only unit_of_measurement when you use a number with value_min, value_max
